src/main/kotlin/com/sourcegraph/utils/CodyEditorUtil.kt

```diff
@@ -184,7 +184,7 @@ object CodyEditorUtil {
                 selection.start.line.toInt(),
                 /* logicalColumn= */ selection.start.character.toInt())
           }
-      runInEdtAndGet { descriptor.navigate(/* requestFocus= */ preserveFocus != true) }
+      invokeLater { descriptor.navigate(/* requestFocus= */ preserveFocus != true) }
```

Comment:
I'm wondering if that won't eat the exception, as it will be already executing on the other thread.

Reply:
afaik the exceptions are propagated b/w threads. we are `OpenFileDescriptor` (constructor) requires BGT while navigate requires EDT. the problem was that `runInEdtAndGet` uses different modality than `invokeLater`

Reply:
I verified the links. I tested Open Cody Settings Editor, Generate Unit Tests, Apply (creating new files) - all worked for me without exceptions.

Reply:
I do not see how it could work.
`invokeLater` is executing in other thread, while current one can even end in the meantime.
There is no way `invokeLater` could throw exception back in the current thread without blocking.

Reply:
You can programatically throw exception inside that `invokeLater` block and then call `Generate Unit Tests` and see what the behaviour is?

Reply:
Having this:
<img width="627" alt="image" src="https://github.com/user-attachments/assets/7847f278-2be8-469b-bd93-0c7dce01b989">

I got an internal error:
<img width="1265" alt="image" src="https://github.com/user-attachments/assets/401b65fa-ebff-40fd-86d8-8a958329e653">

Reply:
Fixed.

I verified the links. 
I tested 
- Open Cody Settings Editor, 
- Generate Unit Tests, 
- Apply (creating new files)
- Save as file 

all worked for me without exceptions 🟢